src/datasources/asset/AssetDataSource.test.ts

```diff
@@ -366,6 +366,19 @@ describe('queries', () => {
     await expect(ds.query(buildMetadataQuery(assetMetadataQueryMock))).rejects.toThrow()
   })
 
+  describe('queryReturnType', () => {
+    it('should return default QueryReturnType.AssetId', () => {
+      const returnType = ds.getQueryReturnType();
+      expect(returnType).toBe(QueryReturnType.AssetIdentification);
+    });
+
+    it('should set and get QueryReturnType correctly', () => {
+      ds.setQueryReturnType(QueryReturnType.AssetId);
+      const returnType = ds.getQueryReturnType();
+      expect(returnType).toBe(QueryReturnType.AssetId);
+    });
+  });
+
```

Comment:
Is there a reason you are not adding a test that covers the return of metricFindQuery in case of AssetId return type?